src/frontend/src/pages/PIAListPage/index.tsx

```diff
@@ -30,27 +34,65 @@ const PIAList = () => {
   const [searchText, setSearchText] = useState(
     searchParams.get('searchText') || '',
   );
+  const setSearchParamsForSearchText = () => {
+    const params: any = {};
+
+    params.searchText = searchText;
+    if (searchParams.get('filterPiaDrafterByCurrentUser'))
+      params.filterPiaDrafterByCurrentUser = filterPiaDrafterByCurrentUser;
+    if (searchParams.get('filterByStatus'))
+      params.filterByStatus = filterByStatus;
+    if (searchParams.get('filterByMinistry'))
+      params.filterByMinistry = filterByMinistry;
+
+    for (const [key, value] of searchParams.entries()) {
+      params[key] = value;
+    }
+    setSearchParams(params);
+  };
 
   const updateSearchUrl = () => {
-    navigate(`?searchText=${searchText}`);
+    setSearchParamsForSearchText();
   };
 
   useEffect(() => {
     const listener = (event: any) => {
       if (event.code === 'Enter' || event.code === 'NumpadEnter') {
         event.preventDefault();
-        navigate(`?searchText=${searchText}`);
+        const params: any = {};
+
+        params.searchText = searchText;
```

Comment:
why not conditional here like for other searchParams?

`if (searchParams.get('searchText')) params.searchText = searchText;`